py/picca/delta_extraction/expected_fluxes/dr16_expected_flux.py

```diff
@@ -236,47 +302,37 @@ def _initialize_variables(self):
         #self.log_lambda_var_func_grid = Forest.log_lambda_grid[::int(resize)]
         #end of commented block
 
+    def _initialize_variance_functions(self):
+        """Initialize variance functions
+        The initialized arrays are:
+        - self.get_eta
+        - self.get_fudge
+        - self.get_num_pixels
+        - self.get_valid_fit
+        - self.get_var_lss
+        """
         # if use_ivar_as_weight is set, eta, var_lss and fudge will be ignored
         # print a message to inform the user
         if self.use_ivar_as_weight:
             self.logger.info(("using ivar as weights, ignoring eta, "
                               "var_lss, fudge fits"))
-            eta = np.ones(self.num_bins_variance)
-            var_lss = np.zeros(self.num_bins_variance)
-            fudge = np.zeros(self.num_bins_variance)
-            num_pixels = np.zeros(self.num_bins_variance)
-            valid_fit = np.ones(self.num_bins_variance)
+            valid_fit = np.ones(self.num_bins_variance, dtype=bool)
         # if use_constant_weight is set then initialize eta, var_lss, and fudge
         # with values to have constant weights
         elif self.use_constant_weight:
             self.logger.info(("using constant weights, ignoring eta, "
                               "var_lss, fudge fits"))
-            eta = np.zeros(self.num_bins_variance)
-            var_lss = np.ones(self.num_bins_variance)
-            fudge = np.zeros(self.num_bins_variance)
-            num_pixels = np.zeros(self.num_bins_variance)
             valid_fit = np.ones(self.num_bins_variance, dtype=bool)
         # normal initialization: eta, var_lss, and fudge are ignored in the
         # first iteration
         else:
-            eta = np.ones(self.num_bins_variance)
-            var_lss = np.zeros(self.num_bins_variance) + 0.2
-            fudge = np.zeros(self.num_bins_variance)
-            num_pixels = np.zeros(self.num_bins_variance)
             valid_fit = np.zeros(self.num_bins_variance, dtype=bool)
+            self.fit_variance_functions = ["var_lss"]
```

Comment:
Left over from when I was creating the initialize_get_X functions. It should not be here (otherwise the FixVarlss class won't work. I'll add a commit to fx it